python/ray/serve/_private/replica.py

```diff
@@ -291,6 +298,8 @@ async def handle_request_streaming(
                 request_metadata, request_args, request_kwargs
             )
 
+        sys_msg = {"accepted": True, "queue_len": queue_len}
+        yield pickle.dumps(sys_msg)
```

Comment:
yield a system message with the current queue length before yielding actual user messages

Reply:
Can we do it as a last message, rather than the first?

So that 
 - We don't delay payload's streaming
 - Queue length might be a bit more fresh (depending on how long we're handling it)

Reply:
I agree this would probably be better, but unfortunately this would require some kind of API change because we aren't able to "hook" into the final message (we hand the generator back to the user after scheduling). Could add some kind of Ray Core API for this, but it's pretty hacky.

Reply:
Understood. Ideally, we should try to delineate data and metadata channels though.